### ISSUE TITLE
tests fail

### ISSUE THREAD

**Issue by kenshyx** [Arachnid/solidity-stringutils#5]
Hello :D, I've noticed that one test fails `testCompare` after updating to `solc 0.4.3/dapple 0.8.15`. Also I'm getting a lot of warnings like these. 

``` Testing...
_0xb3668356078994a35ee131f8ba3e76d3441e5f52056a912f94278c22ff495034.sol:55:18: Warning: Inline assembly block is not balanced. It leaves 2 item(s) on the stack.
        assembly {
                 ^
Spanning multiple lines.
_0xb3668356078994a35ee131f8ba3e76d3441e5f52056a912f94278c22ff495034.sol:115:18: Warning: Inline assembly block is not balanced. It leaves 1 item(s) on the stack.
        assembly {
                 ^
...
StringsTest
  test compare
  Error at `e only h'
```

**Comment by simondlr** [comment]
Getting the same with truffle Solidity tests.

**Comment by simondlr** [comment]
fwiw. I made a work-around to this in Truffle by making it ignore these warnings for now.

Ideally, the code should be update so that it doesn't throw these warnings. I don't understand these warnings though. I need to first read up on how to fix this before I can submit a PR. :)

**Comment by simondlr** [comment]
This was an error in the Solidity compiler. It was fixed in solc 0.4.5.
